extensions/parquet/base/src/main/java/io/deephaven/parquet/base/ParquetFileReader.java

```diff
@@ -11,43 +12,79 @@
 import org.apache.parquet.schema.*;
 
 import java.io.ByteArrayInputStream;
+import java.io.File;
 import java.io.IOException;
+import java.net.URI;
+import java.net.URISyntaxException;
 import java.nio.ByteBuffer;
 import java.nio.ByteOrder;
 import java.nio.channels.SeekableByteChannel;
 import java.nio.charset.StandardCharsets;
-import java.nio.file.Path;
-import java.nio.file.Paths;
 import java.util.*;
 
 /**
- * Top level accessor for a parquet file
+ * Top level accessor for a parquet file which can read both from a file path string or a CLI style file URI,
+ * ex."s3://bucket/key".
  */
 public class ParquetFileReader {
     private static final int FOOTER_LENGTH_SIZE = 4;
     private static final String MAGIC_STR = "PAR1";
     static final byte[] MAGIC = MAGIC_STR.getBytes(StandardCharsets.US_ASCII);
+    public static final String S3_URI_SCHEME = "s3";
+    private static final String S3_URI_PREFIX = "s3://";
 
     public final FileMetaData fileMetaData;
     private final SeekableChannelsProvider channelsProvider;
-    private final Path rootPath;
+
+    /**
+     * If reading a single parquet file, root URI is the URI of the file, else the parent directory for a metadata file
+     */
+    private final URI rootURI;
     private final MessageType type;
 
+    // TODO Where should I keep it?
+    public static URI convertToURI(final String filePath) {
```

Comment:
This impl is not very good. For example, if somebody tries to pass an explicit file URI, `file:///tmp/my-file.parquet` this will be treated as a *relative* path to something that is very likely not what the user meant:

```
jshell> new File("file:///tmp/my-file.parquet").getAbsoluteFile().toURI()
$11 ==> file:/home/devin/dev/deephaven/deephaven-core/file:/tmp/my-file.parquet
```

The logic probably needs to be something that looks like 1) create URI; 2) if has scheme, return URI; 3) if doesn't have scheme, fallback to `new File(filePath).getAbsoluteFile().toURI()`

Reply:
I have made the change but the challenge with the new design is that if there is a file with spaces or special characters in the name, users will have to URI encode the name.
For example, file name "table name.parquet" will have to be provided as "table%20name.parquet".
Do you think that's okay?